features.json

```diff
@@ -0,0 +1,123 @@
+{
+	"features": {
+		"bigInt": {
+			"description": "JavaScript BigInt to WebAssembly i64 integration",
+			"url": "https://github.com/WebAssembly/JS-BigInt-integration",
+			"phase": 3
+		},
+		"bulkMemory": {
+			"description": "Bulk memory operations",
+			"url": "https://github.com/WebAssembly/bulk-memory-operations/blob/master/proposals/bulk-memory-operations/Overview.md",
+			"phase": 3
+		},
+		"exceptions": {
+			"description": "Exception handling",
+			"url": "https://github.com/WebAssembly/exception-handling/blob/master/proposals/Exceptions.md",
+			"phase": 2
+		},
+		"multiValue": {
+			"description": "Multi-value",
+			"url": "https://github.com/WebAssembly/spec/blob/master/proposals/multi-value/Overview.md",
+			"phase": 4
+		},
+		"mutableGlobals": {
+			"description": "Import & export of mutable globals",
+			"url": "https://github.com/WebAssembly/mutable-global/blob/master/proposals/mutable-global/Overview.md",
+			"phase": 4
+		},
+		"referenceTypes": {
+			"description": "Reference types",
+			"url": "https://github.com/WebAssembly/reference-types/blob/master/proposals/reference-types/Overview.md",
+			"phase": 3
+		},
+		"saturatedFloatToInt": {
+			"description": "Non-trapping float-to-int conversions",
+			"url": "https://github.com/WebAssembly/spec/blob/master/proposals/nontrapping-float-to-int-conversion/Overview.md",
+			"phase": 4
+		},
+		"signExtensions": {
+			"description": "Sign-extension operations",
+			"url": "https://github.com/WebAssembly/spec/blob/master/proposals/sign-extension-ops/Overview.md",
+			"phase": 4
+		},
+		"simd": {
+			"description": "Fixed-width SIMD",
+			"url": "https://github.com/WebAssembly/simd/blob/master/proposals/simd/SIMD.md",
+			"phase": 3
+		},
+		"tailCall": {
+			"description": "Tail calls",
+			"url": "https://github.com/WebAssembly/tail-call/blob/master/proposals/tail-call/Overview.md",
+			"phase": 3
+		},
+		"threads": {
+			"description": "Threads and atomics",
+			"url": "https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md",
+			"phase": 2
+		}
+	},
+	"browsers": {
+		"Chrome": {
+			"url": "https://www.google.com/chrome/",
+			"logo": "/images/chrome.svg",
+			"features": {
+				"bigInt": "#enable-experimental-webassembly-features",
```

Comment:
This is useful, but I'm worried it will get out of date too. I guess it doesn't change that often, though.

Reply:
Well that's the point of caniuse-like table - to show feature support across engines. Indeed, it will require constant maintenance, but it's also currently the only way to find out which Wasm features are supported in chosen browsers and so on.

Reply:
> Indeed, it will require constant maintenance, but it's also currently the only way to find out which Wasm features are supported in chosen browsers and so on.

I do agree that this is useful to have, but I share @binji's worries about it becoming outdated quickly. Can you share more information on your plans for keeping it up to date? (Case in point, in Wasmtime we recently [enabled multi-value by default](https://github.com/bytecodealliance/wasmtime/pull/1667), [finished implementing bulk memory](https://github.com/bytecodealliance/wasmtime/issues/928), and are [actively working on reference types](https://github.com/bytecodealliance/wasmtime/issues/929).

Reply:
Just like on caniuse.com, MDN or feature tables in https://v8.dev/features, I expect that this will be updated by website maintainers or engine maintainers relatively quickly after features are implemented. All of these websites share similar concerns, but on practice delay has never been too big thanks to community contributions.

I'm happy to update this table (admittedly, I wrote this JSON about a ~month ago and only made a PR now) with new Wasmtime features.

Reply:
> Just like on caniuse.com, MDN or feature tables in https://v8.dev/features, I expect that this will be updated by website maintainers or engine maintainers relatively quickly after features are implemented.

I've seen this go wrong a good number of times, so I'm not sure this is a valid assumption to make. Would it make sense to bring this up in the CG and ensure that there's some amount of commitment from enough of the relevant implementers to really do this before landing this?

Reply:
What is the risk scenario here? In my perspective, tables like this are useful for development to know when you can compile with a specific feature or when you need to ship a polyfill / a version that doesn't rely on certain new feature.

Even if tables gets outdated, worst case is a false negative, which means that someone unnecessarily ships a polyfill to the engine that has a native support now, which might be suboptimal, but not critical by any means.

What matters / is the goal here is the best-case scenario where someone can ship a more optimised version (e.g. using SIMD or threads or native integer conversion ops) to engines that definitely support it, and for that scenario even an outdated table is better than nothing (what we have right now).

---

All in all, I'm commited to keeping this reasonably up-to-date, but I don't see delays as a critical problem for the original goal which is unblocking developers in using / testing new features on "definitely supported" platforms.

Reply:
Agreed, it's probably better to have out-of-date information than to have no information here. Maybe as a compromise we can include version numbers for the tools? Or even simpler, we could include a "last updated" mention.

Reply:
"Last updated" sounds like a good idea, yeah.

Reply:
I've updated the PR with version numbers for each engine so that it's immediately clear what version the table was tested against and reflects; please see a screenshot below and let me know what you think.